File: Medium/238 Product of Array/bruteforce.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    vector<int> productExceptSelf(vector<int> &nums)
    {
        vector<int> ans = {};

        for (int i = 0; i < nums.size(); i++)
        {
            int product = 1;
            for (int j = 0; j < nums.size(); j++)
            {
                if (i == j)
                    continue;

                product *= nums[j];
            }

            ans.push_back(product);
        }

        return ans;
    }
};

int main()
{

    Solution sol;
    vector<int> nums = {};

    for (int i = 0; i < 1000000; i++)
    {
        nums.push_back(3);
    }

    sol.productExceptSelf(nums);
}

File: Medium/238 Product of Array/bruteforce.py
class Solution:
    def productExceptSelf(self, nums: list[int]) -> list[int]:
        ans = [0]*len(nums)
        for i in range(len(nums)):
            product = 1
            for j in range(len(nums)):
                if i == j:
                    continue
                product *= nums[j]
            ans[i] = product
        return ans
    
sol = Solution()

nums = [1,2,3,4]
nums = [-1,1,0,-3,3]
nums = [3]*10000
print(len(nums))
ans = sol.productExceptSelf(nums)

# O(n^2)